### PR TITLE
Math magicians: Events

### PR DESCRIPTION
**Pull request by Kalunge**
## Math magicians: Events
in this milestone I:
- Implemented the event handlers needed to use the math logic from calculator.js and operate.js in your Calculator component.
- Tested the app in the browser, it is able to perform math operations.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '÷') {
+    return one.div(two).toString();
+  }
```
Comment (DeliceLydia) [inline src/logic/operate.js]: - [ ] Please work on the division of a number with zero let us say `9/0` it should return `error` in the screen but currently its returning
shot
please work on that

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
```
Comment (DeliceLydia) [inline src/logic/calculate.js]: - [ ] I recommend that you add a default `zero` in the screen when you press `AC` and consider Adding a default zero in the screen when you just opened the app

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Kalunge,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.js (hunk 2 above)
Comment (Kalunge) [reply src/logic/calculate.js]: Thanks for pointing this out. I have done the necessary changes

Comment (henatan99) [reply src/logic/calculate.js]: This issue is well addressed! :+1: :clap:

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '÷' && numberTwo !== '0') {
+    return one.div(two).toString();
+  }
```
Comment (henatan99) [inline src/logic/operate.js:17]: - [ ] The issue of a number divided by zero was mentioned by the previous reviewer. Yet you did not make a change. You may not necessarily make changes in the `operate.js` as it was generated by Microverse. You can also add error handling here or in your `Calculator.js` file. Kindly make sure that the error is well handled. 

Screenshot from 2021-09-15 17-09-00

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Kalunge , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/operate.js (hunk 3 above)
Comment (Kalunge) [reply src/logic/operate.js:17]: But as you can see the error emanates from the Microverse code. How am I to resolve it without touching the microverse code.

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import Big from 'big.js';
+
+export default function operate(numberOne, numberTwo, operation) {
+  const one = Big(numberOne);
+  const two = Big(numberTwo);
+  if (operation === '+') {
+    return one.plus(two).toString();
+  }
+  if (operation === '-') {
+    return one.minus(two).toString();
```
Comment (Dipeshtwis) [inline src/logic/operate.js:10]: [OPTIONAL]

- When I tried to do something like `8 +` and then I pressed any operation again like `-`, it's better if it's changed into `8 -` removing the previous operation. But in your case, it's saying invalid which shows in the below screenshot. It's better if you work on that after that your app will be fully bug-free.

Screenshot from 2021-09-15 18-55-24

**Review by Dipeshtwis — APPROVED** [review]
# Hi @Kalunge, 

### Status: Approved ✔️

:+1:  You have implemented the changes requested by the previous reviewer. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**_Check the comments under the review_**



**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/operate.js (hunk 4 above)
Comment (Kalunge) [reply src/logic/operate.js:10]: Thanks, will definitely work on that

**Comment by SpaYco** [comment]
Hello 👋,

_This is an additional review as a result of the second opinion request._

I'm happy that your issues resolved, but we need to make sure that this won't happen again, can you create an issue on the repo where this file is, and explain in detail what's the problem, and we'll fix it ASAP!

Keep rocking 🚀!

**Pull request merged**